reconcile/utils/terraform_resource_spec.py

```diff
@@ -0,0 +1,123 @@
+from dataclasses import dataclass, field
+import json
+from typing import Any, Optional, cast
+import base64
+from reconcile.utils.openshift_resource import OpenshiftResource as OR
+
+
+@dataclass
+class TerraformResourceSpec:
+
+    resource: dict[str, Any]
+    namespace: dict[str, Any]
+    secret: dict[str, str] = field(init=False, default_factory=lambda: {})
+
+    @property
+    def provider(self):
+        return self.resource.get("provider")
+
+    @property
+    def identifier(self):
+        return self.resource.get("identifier")
+
+    @property
+    def account(self):
+        return self.resource.get("account")
+
+    @property
+    def namespace_name(self) -> str:
+        return self.namespace["name"]
+
+    @property
+    def cluster_name(self) -> str:
+        return self.namespace["cluster"]["name"]
+
+    @property
+    def output_prefix(self):
+        return f"{self.identifier}-{self.provider}"
+
+    @property
+    def output_resource_name(self):
+        return self.resource.get("output_resource_name") or self.output_prefix
+
+    @property
+    def annotations(self) -> dict[str, str]:
+        annotation_str = self.resource.get("annotations")
+        if annotation_str:
+            return json.loads(annotation_str)
+        else:
+            return {}
+
+    def get_secret_field(self, field: str) -> Optional[str]:
+        return self.secret.get(field)
+
+    def id_object(self) -> "TerraformResourceIdentifier":
```

Comment:
why the stringified return type hint?

Reply:
that is a forward reference because the type is later defined in the file

Reply:
ok! tjx